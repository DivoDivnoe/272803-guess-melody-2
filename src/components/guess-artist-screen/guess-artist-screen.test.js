import React from 'react';
import renderer from 'react-test-renderer';
import GuessArtistScreen from './guess-artist-screen.jsx';

jest.mock(`../player/player.jsx`, () => jest.fn().mockReturnValue(null));
jest.mock(`../lifes/lifes.jsx`, () => jest.fn().mockReturnValue(null));

describe(`GuessArtistScreen component`, () => {
  it(`is rendered correctly`, () => {
    const question = {
      type: `artist`,
      song: {artist: ``, src: ``},
      answers: [{picture: `http://somesrc/`, artist: ``}]
    };
    const screenIndex = 0;

    const tree = renderer.create(
        <GuessArtistScreen
          question={question}
          screenIndex={screenIndex}
          onClick={jest.fn()}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
