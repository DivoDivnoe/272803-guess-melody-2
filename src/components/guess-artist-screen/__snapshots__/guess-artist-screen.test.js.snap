// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuessArtistScreen component is rendered correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      />
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="\${answer.artist}"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt=""
            className="artist__picture"
            src="http://somesrc/"
          />
          
        </label>
      </div>
    </form>
  </section>
</section>
`;
